TTSDiceCalculator.lua

```diff
@@ -15,30 +14,27 @@ function newPlayerRoll(steam_name)
     local self = {
         dice = {},
         steam_name = steam_name,
+        waitID = nil
     }
 
-    local still_rolling = function()
-        Timer.destroy(self.steam_name .. ":roll_timer")
-        Timer.create({
-            identifier = self.steam_name .. ":roll_timer",
-            function_name = "finishRoll",
-            parameters = { self.steam_name },
-            delay = DELAY_TIME, --in seconds
-        })
-    end
-
     local add = function(object)
-        still_rolling()
         self.dice[object.getGUID()] = object
-    end
-
-    local really_finished = function()
-        for k, v in pairs(self.dice) do
-            if not v.resting then
-                return false
-            end
+        -- After first add, wait 0.5s and start a wait for all dice to rest
+        if not self.waitID then
+            self.waitID = Wait.time(function()
```

Comment:
Half second wait is just to make sure the newly added die is already in motion. TTS can fire the onObjectRandomize trigger before physics kick in and the die still reads as obj.resting, we don;t want this to happen and roll finish condition to trigger immediately.